### DISCUSSION TITLE
Bluetooth sink and source at the same time

### DISCUSSION THREAD

**Discussion by Quirin9** [pschatzmann/ESP32-A2DP#286] [Q&A] [answered]
I am working on a spectrum analyzer project with the esp32 and want to stream audio to the esp32 from a phone then do a fft on the esp32. At the same time the esp32 has to stream the same audio to a bluetooth speaker.
Is it possible to make the esp a source and a sink at the same time?
Thank you for your help!

**Comment by pschatzmann** [comment, marked as answer]
No, that will not work. But you can implement your own bluetooth speaker with the A2DP sink and do the FFT there.
Another option would be to simply get the audio via a microphone..

**Reply by degnarraer** [reply]
2x ESP32s linked together via i2s. ESP32 #1 is your sink. ESP32 #2 is your source. Audio data is passed from sink to source via i2s link between the 2 ESP32s. I am currently doing this exactly thing in my project.
